### PR TITLE
Support NPU kernel stack op

### PR DESCRIPTION
**Pull request by oyxuan-11**
### PR types
New features

### PR changes
OPs

### Describe
【NPU】Support npu op stack

单测通过截图
image

调用NPU截图
image

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+~
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py]: Delete line 1

File: python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    def set_npu(self):
+        self.__class__.use_npu = True
+
+    def test_check_output(self):
+        self.check_output_with_place(self.place, check_dygraph=False)
+
+
+class TestStack2(OpTest):
+
+    def initDefaultParameters(self):
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py:73]: Please use pre-commit to check code.

File: python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py (hunk 1 above)
Comment (oyxuan-11) [reply python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py]: fixed

File: python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py (hunk 2 above)
Comment (oyxuan-11) [reply python/paddle/fluid/tests/unittests/npu/test_stack_op_npu.py:73]: fixed

**Review by liym27 — APPROVED** [review]
LGTM

File: paddle/fluid/operators/stack_op_npu.cc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    for (int i = 0; i < n; i++) {
+      x_list.push_back(*x[i]);
+    }
+
+    int axis = ctx.Attr<int>("axis");
+
+    if (axis < 0) {
+      axis = axis + x_list[0].dims().size() + 1;
+    }
+    int32_t N = x.size();
```
Comment (pangyoki) [inline paddle/fluid/operators/stack_op_npu.cc]: 这里的`N`与前面的`n`有什么区别？能否只保留其中一个

Comment (oyxuan-11) [reply paddle/fluid/operators/stack_op_npu.cc]: fixed

**Review by pangyoki — APPROVED** [review]
LGTM

**Pull request merged**